state-chain/pallets/cf-elections/src/electoral_systems/block_witnesser/block_processor.rs

```diff
@@ -190,14 +189,15 @@ impl<T: BWProcessorTypes> BlockProcessor<T> {
 		last_height: T::ChainBlockNumber,
 	) -> Vec<(T::ChainBlockNumber, T::Event)> {
 		let mut last_events: Vec<(T::ChainBlockNumber, T::Event)> = vec![];
-		for (block_height, (data, next_age)) in self.blocks_data.clone() {
+		for (block_height, (data, next_age_to_process)) in self.blocks_data.clone() {
 			let current_age = T::ChainBlockNumber::steps_between(&block_height, &last_height).0;
-			for age in next_age..=current_age as u32 {
-				last_events = last_events
-					.into_iter()
-					.chain(self.process_rules_for_age_and_block(block_height, age, &data))
-					.collect();
-			}
+			let age_range: RangeInclusive<u32> =
+				RangeInclusive::new(next_age_to_process, current_age as u32);
```

Comment:
nit: I think you can use the `a..=b` notation to construct `RangeInclusive` objects. (And `a..b` for `Range`s of course).